### PR TITLE
linux-v4l2: Expand packed values to 4 bytes

### PR DESCRIPTION
**Pull request by kkartaltepe**
### Description

Expand packed values to 4 bytes, and wrap up static values with a macro.

### Motivation and Context

These days you might capture card a 175Hz monitor with a framerate defined as 10000000/57143 that doesnt fit in a short, as reported by a user. And reporting a negative FPS in that case isnt very helpful.

### How Has This Been Tested?

My logitech webcam still works.

### Types of changes

- Bug fix (non-breaking change which fixes an issue)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by kkartaltepe** [comment]
Actually this needs more work, we need to store resolution and framerate somewhere else and upgrade the old properties...

File: plugins/linux-v4l2/v4l2-helpers.h [hunk 1]
```diff
@@ -91,24 +94,24 @@ static inline enum video_format v4l2_to_obs_video_format(uint_fast32_t format)
  * and the height in the low word.
  * The array is terminated with a zero.
  */
-static const int v4l2_framesizes[] = {
+static const uint64_t v4l2_framesizes[] = {
```
Comment (derrod) [inline plugins/linux-v4l2/v4l2-helpers.h:97]: Why is this a `uint64_t` instead of `int64_t` as elsewhere in packed numbers?

**Review by derrod — COMMENTED** [review]
> Actually this needs more work, we need to store resolution and framerate somewhere else and upgrade the old properties...

Sounds like you either migrate by changing the key to a new one and copy from the old one if it exists, or figure out a way to detect the old value's format.

File: plugins/linux-v4l2/v4l2-helpers.h (hunk 1 above)
Comment (kkartaltepe) [reply plugins/linux-v4l2/v4l2-helpers.h:97]: Fixed (in a second), it should be an int64_t.

**Comment by kkartaltepe** [comment]
> > Actually this needs more work, we need to store resolution and framerate somewhere else and upgrade the old properties...
> 
> Sounds like you either migrate by changing the key to a new one and copy from the old one if it exists, or figure out a way to detect the old value's format.

I slept on it, and it should be safe to just treat any packed value with no high bits as the old 32bit form.

**Review by Lain-B — APPROVED** [review]

**Pull request merged**